### ISSUE TITLE
async imports are not parsed correctly

### ISSUE THREAD

**Issue by mx-bernhard** [Dschungelabenteuer/vite-plugin-entry-shaking#37]
**Describe the bug**

When importing from an entry file using an async import, the import statement ends up with an invalid js syntax:

```js
const { a, b } = await import('./my-entry');
```

**To Reproduce**

See my [repro](https://github.com/mx-bernhard/vite-plugin-entry-shaking/tree/issue-async-import).

    cd ./examples/simple
    pnpm run dev
    Open browser on location
    See error browser console:
    Uncaught SyntaxError: Unexpected token ';' (at main.ts:2:24)

**Expected behavior**

Maybe, the async import statement is left as is because it will be loaded deferred anyway. It is probably harder to analyze, and I wouldn't bother.

**Additional context**

The issue comes from this regex in the implementation in _analyze-import.ts_:

```js
  const [, importContentString] =
    code.slice(startPosition, endPosition).match(/{([\s\S]*?)}/) ?? [];
```

This doesn't match because the curly braces are missing here. An empty array is returned instead. I would suggest to check for an empty array on the caller of this function like this:

```js
  const namedImports = methods.getImportedNamedExports(code, startPosition, endPosition);
  if (namedImports.length > 0) {
    const imported = await methods.getImportsMap(entryExports, entryPath, namedImports, resolver);
    const replacement = await methods.getImportReplacements(imported, entryPath, entries, resolver);

    src.overwrite(startPosition, endPosition + 1, `${replacement.join(';\n')};`);
  }
```

But maybe this causes new problems, can't tell yet. It works in our (bigger) project.

**Comment by Dschungelabenteuer** [comment]
Just realized I did not investigate this before publishing 0.4.0, apologies! I'll take a look soon!

**Comment by Dschungelabenteuer** [comment]
Sorry it took so long for me to address this!

>  I would suggest to check for an empty array on the caller of this function (...)

Well, I've came up with the very same conclusion. As stated in the limitations part of the README, dynamic/async imports are not supported but still shouldn't break (it should just prevent tree-shaking), yet it is so you've got a fair point :')  

Judging by the quality of your bug report and your other contributions, I gather you did your research and probably noticed how `es-module-lexer` only focuses on the actual import statement, rather than the whole `VariableDeclaration` node. This prevents any efficient application of this plugin's logic to such statement as we know the imported path, but not the imported entities from that path. Basically, we can't expect tree-shaking to be working here and should just skip those statement untouched so that it doesn't crash. Your suggestion definitely is the way to go.

> But maybe this causes new problems, can't tell yet.

Well, this sole fix could still cause issues in some cases. 

This plugin tries as much as possible not to request any entry file so that it doesn't end up unnecessarily requesting their whole subsequent dependencies. Still, some entry files could be actually **defining and exporting** code that could be imported from any other file. Therefore, we maintain a cleaned-up version of such entry files so that we can still serve them without triggering too much requests.

Now, solely ignoring any dynamic/async import of an entry file would cause Vite dev server to actually serve its cleaned-up version. And that cleaned-up version may not include the entity being imported: error! Simple example to illustrate my point:

```ts
// ./entry/index.ts (original)
export { a } from './a';
export const b = 'b';

// ./entry/index.ts (cleaned-up)
export const b = 'b';

// ./consumer.ts
const { a, b } = await import ('./entry');
//      ^ error: a is undefined (since cleaned-up and would be re-written to `./entry/a.ts` in normal imports)
```

Good news is, this plugin already supports a `source` query string that allows us to serve the original module instead of the cleaned-up one. In other words, we only have to transform dynamic/async imports like this:

```ts
// ./consumer.ts
const { a, b } = await import ('./entry?source=1');
```

I'll probably release a fix very soon ;)

**Comment by Dschungelabenteuer** [comment]
It should be fixed with 0.4.3, please feel free to re-open if needed :-)
